pyerrors/input/sfcf.py

```diff
@@ -29,266 +53,210 @@ def read_sfcf(path, prefix, name, **kwargs):
             deltas = []
             for j in range(T):
                 deltas.append([])
-
+        
         sublength = no_cfg
         for j in range(T):
             deltas[j].append(np.zeros(sublength))
-
-        for cfg in range(no_cfg):
-            with open(path + '/' + item + '/' + sub_ls[cfg]) as fp:
-                for k, line in enumerate(fp):
-                    if k == start_read - 5 - b2b:
-                        if line.strip() != 'name      ' + name:
-                            raise Exception('Wrong format', sub_ls[cfg])
-                    if(k >= start_read and k < start_read + T):
-                        floats = list(map(float, line.split()))
-                        deltas[k - start_read][i][cfg] = floats[-2:][im]
-
+        #... the actual parsing can start. we iterate through all measurement files in the path given...
+        if compact:
+            for cfg in range(no_cfg):
+                with open(path + '/' + item + '/' + sub_ls[cfg]) as fp:
+                    lines = fp.readlines()
+                    #check, if the correlator is in fact printed completely
+                    if(start_read + T>len(lines)):
+                        raise Exception("EOF before end of correlator data! Maybe "+path + '/' + item + '/' + sub_ls[cfg]+" is corrupted?")
+                    #and start to read the correlator.
+                    #the range here is chosen like this, since this allows for implementing a security check for every read correlator later...
+                    for k in range(start_read - 6,start_read + T):
+                        if k == start_read - 5 - b2b:
+                            if lines[k].strip() != 'name      ' + name:
+                                raise Exception('Wrong format', sub_ls[cfg])
+                        if(k >= start_read and k < start_read + T):
+                            floats = list(map(float, lines[k].split()))
+                            deltas[k - start_read][i][cfg] = floats[-2:][im]
+        else:
+            for cnfg, subitem in enumerate(sub_ls):
+                with open(path + '/' + item + '/' + subitem + '/' + name) as fp:
+                    #since the non-compatified files are typically not so long, we can iterate over the whole file.
+                    #here one can also implement the chekc from above.
+                    for k, line in enumerate(fp):
+                        if(k >= start and k < start + T):
+                            floats = list(map(float, line.split()))
+                            if version == "0.0":
+                                deltas[k-start][i][cnfg] = floats[im]
+                            else:
+                                deltas[k - start][i][cnfg] = floats[1 + im - single]
+
+    if "check_configs" in kwargs:
+        print("Chekcing for missing configs...")
+        che = kwargs.get("check_configs")
+        if not (len(che) == len(idl)):
```

Comment:
You could also do something like 
```python
if len(set(che).symmetric_difference(idl)) > 0
```
to catch cases, where the two lists are not equal but have the same length.

Reply:
I think I do not understand this comment. These two are lists of lists in which the configurations per replicum are listed, this part just checks if there is a list to check for each replicum. `set(che).symmetric_difference(idl)` is almost what is done here, together with `check _idl()` for each rep separately.